compiler/src/main/java/com/rejasupotaro/android/kvs/internal/SchemaWriter.java

```diff
@@ -21,129 +25,144 @@ public SchemaWriter(SchemaModel model) {
 
     public void write(Filer filer) {
         try {
-            StringBuilder fqcn = new StringBuilder();
-            fqcn.append(model.getPackageName())
-                    .append(".")
-                    .append(model.getClassName());
-
-            JavaFileObject sourceFile = filer.createSourceFile(fqcn.toString(), model.getElement());
-            JavaWriter writer = new JavaWriter(sourceFile.openWriter());
-
-            writer.emitPackage(model.getPackageName());
-            writeImports(writer);
-            writer.beginType(model.getClassName(), "class", EnumSet.of(Modifier.PUBLIC, Modifier.FINAL), model.getOriginalClassName());
+            TypeSpec.Builder classBuilder = TypeSpec.classBuilder(model.getClassName());
+            classBuilder.addModifiers(Modifier.PUBLIC, Modifier.FINAL);
+            ClassName superClassName = ClassName.get(model.getPackageName(), model.getOriginalClassName());
+            classBuilder.superclass(superClassName);
+
+            List<FieldSpec> fieldSpecs = createFields();
+            for (FieldSpec fieldSpec : fieldSpecs) {
+                classBuilder.addField(fieldSpec);
+            }
 
-            writeFields(writer);
-            writeConstructors(writer);
-            writeMethods(writer);
+            List<MethodSpec> methodSpecs = new ArrayList<>();
+            methodSpecs.addAll(createConstructors());
+            methodSpecs.addAll(createMethods());
+            for (MethodSpec methodSpec : methodSpecs) {
+                classBuilder.addMethod(methodSpec);
+            }
 
-            writer.endType();
-            writer.close();
+            TypeSpec outClass = classBuilder.build();
+            JavaFile.builder(model.getPackageName(), outClass)
+                    .build()
+                    .writeTo(filer);
         } catch (IOException e) {
             e.printStackTrace();
```

Comment:
This is out of scope of this PR, but it would be better to report errors from Messager. https://docs.oracle.com/javase/8/docs/api/javax/annotation/processing/Messager.html